fenix/app/src/main/java/org/mozilla/fenix/extension/WebExtensionPromptFeature.kt

```diff
@@ -108,10 +112,32 @@ class WebExtensionPromptFeature(
         addon: Addon,
         promptRequest: WebExtensionPromptRequest.AfterInstallation.Permissions,
     ) {
-        if (hasExistingPermissionDialogFragment()) return
         showPermissionDialog(
-            addon,
-            promptRequest,
+            addon = addon,
+            onConfirm = promptRequest.onConfirm,
+        )
+    }
+
+    @VisibleForTesting
+    internal fun handleOptionalPermissionsRequest(
+        addon: Addon,
+        promptRequest: WebExtensionPromptRequest.AfterInstallation.OptionalPermissions,
+    ) {
+        val shouldGrantWithoutPrompt = Addon.localizePermissions(promptRequest.permissions, context).isEmpty()
+
+        // If we don't have any promptable permissions, just proceed.
+        if (shouldGrantWithoutPrompt) {
+            promptRequest.onConfirm(true)
+            consumePromptRequest()
+            return
+        }
+
+        showPermissionDialog(
+            // This is a bit of a hack so that the permission prompt only lists
+            // the optional permissions that are requested.
+            addon = addon.copy(permissions = promptRequest.permissions),
```

Comment:
I think to avoid future issues, we could just add a new property on the prompt for optional permissions, this way we will explicit about each uses of the permissions  :)

Reply:
I considered this but... an add-on can have optional permissions listed in the manifest, and that's what the property on an `Addon` should probably contain.

Now, when the extension invokes `permissions.request()`, it can pass any of these optional permissions, either one, two or all, and this is what `promptRequest.permissions` contains here. That's why I am not sure a new property on `Addon` would be a better approach.

Thinking about it more, maybe the fragment shouldn't receive an `Addon` instance but the name, icon, permissions, etc.?

Reply:
> Thinking about it more, maybe the fragment shouldn't receive an Addon instance but the name, icon, permissions, etc.?

That sounds like a better approach!

Reply:
Just bringing this related comment https://github.com/mozilla-mobile/firefox-android/pull/3917#discussion_r1345938566